pkg/cluster/validation.go

```diff
@@ -78,3 +80,30 @@ func ValidatePartial(cluster *storage.Cluster) *errorhelpers.ErrorList {
 
 	return errorList
 }
+
+func IsNameValid(name string) (bool, error) {
+	// This check is added to avoid problems with rendering of helm templates.
+	// If a number is added as a cluster name, Helm may treat is as such and
+	// convert it to scientific notation - see cluster name in sensor secret.
+	// Because the fix in the Helm chart is tricky, we reject all names that
+	// could be parsable to a number or any other Helm construct.
+	if len(name) == 0 {
+		return false, errors.New("cluster name cannot be empty")
+	}
+	if errs := validation.IsDNS1123Subdomain(name); len(errs) != 0 {
+		return false, errors.New(strings.Join(errs, "; "))
+	}
```

Comment:
@rukletsov this unfortunately disallows underscores in the names - that may be a huge change. We use underscores in our unit tests, but I think we should relax the rules and allow them in the name - WDYT?

Reply:
Moreover, this validation disallows also upper-case characters. I think we must change it

Reply:
Yeah, looks like namespaces are validated via `IsDNS1123Label()`, see [this](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/validation#ValidateNamespaceName) and [this](https://github.com/kubernetes/apimachinery/blob/37988e577e160bf7749437141fee1d6b8ca15dd6/pkg/api/validation/generic.go#L45-L50).

Reply:
Interestingly, upper-case is not allowed there either